### DISCUSSION TITLE
Using typings-for-css-modules-loader (or smth simular) with Next JS

### DISCUSSION THREAD

**Discussion by Passhub** [vercel/next.js#39789] [Help]
Hi everyone!

I was wondering how can we setup Next Js application **with automatic generation of typed d.ts files to css (scss, sass) files**?

For example, we are using module.scss files in out project and really wants to know what is inside this files after we import them.
Looks like this (
![image](https://user-images.githubusercontent.com/31039433/185763730-68559ffe-4b2f-474b-8ef7-cf7f4991e489.png)
)

For that, in **not Next JS** applications we can use additional loader for css files in webpack, for example **typings-for-css-modules-loader**.

I was trying to add additional loader to webpack configuration object in nextjs config file, but it doesn't worked fo me.
Here how it looks like:

```
const nextConfig = {
   webpack: (config, context) => {
      const oneOf = config.module.rules.find(
        (rule) => typeof rule.oneOf === "object"
      );
    
    if (oneOf) {
      // Find the module which targets *.scss|*.sass files
      const moduleSassRule = oneOf.oneOf.find((rule) =>
        regexEqual(rule.test, /\.module\.(scss|sass)$/)
      );

      if (moduleSassRule) {
        moduleSassRule.use.push({
          loader: "@teamsupercell/typings-for-css-modules-loader",
          options: {
            banner:
              "// autogenerated by typings-for-css-modules-loader. \n// Please do not change this file!",
          },
        });
      }
    }

    return config;
  }
}
```

I searched a lot, but didn't find any suggestions...

Maybe someone can share any ideas how to deal with that "feature".

I will be very thankfull for any advise!

Have a good one)

**Comment by ArianHamdi** [comment]
Hey guys, did you find any way to achieve this?

**Comment by dlabaja** [comment]
same question
